scvi/external/sysvi/_model.py

```diff
@@ -0,0 +1,336 @@
+from __future__ import annotations
+
+import logging
+from collections.abc import Sequence
+
+import numpy as np
+import pandas as pd
+import torch
+from anndata import AnnData
+from typing_extensions import Literal
+
+from scvi import REGISTRY_KEYS
+from scvi.data import AnnDataManager
+from scvi.data.fields import (
+    LayerField,
+    ObsmField,
+)
+from scvi.model.base import BaseModelClass, UnsupervisedTrainingMixin
+from scvi.utils import setup_anndata_dsp
+
+from ._module import SysVAE
+from ._trainingplans import TrainingPlanCustom
+from ._utils import prepare_metadata
+
+logger = logging.getLogger(__name__)
+
+
+class TrainingCustom(UnsupervisedTrainingMixin):
+    """Train method with custom TrainingPlan."""
+
+    # TODO could make custom Trainer (in a custom TrainRunner) to have in init params for early stopping
+    #  "loss" rather than "elbo" components in available param specifications - for now just use
+    #  a loss that is against the param specification
+
+    _training_plan_cls = TrainingPlanCustom
```

Comment:
Is it still necessary to have this part of the code? One way around this would be to make `SysVI` inherit from `UnsupervisedTrainingMixin` and then set `_training_plan_cls = TrainingPlanCustom` directly in it.